packages/access-generic-tracers/package.py

```diff
@@ -29,21 +34,36 @@ class AccessGenericTracers(CMakePackage):
         description="If True, depend on access-fms, otherwise depend on fms"
     )
 
-    depends_on("access-mocsy")
-    depends_on("access-fms", when="+use_access_fms")
-    depends_on("fms@2025.02:", when="~use_access_fms")
     depends_on("mpi")
+    # TODO: Make conditional once Spack v0.23 or newer is used. The newer
+    #       versions contain an fms SPR with variant shared.
+    depends_on("fms@2025.02:", when="~use_access_fms")
+    with when("+shared"):
+        depends_on("access-mocsy+shared")
+        depends_on("access-fms+shared", when="+use_access_fms")
+    with when("~shared"):
+        depends_on("access-mocsy")
+        depends_on("access-fms", when="+use_access_fms")
 
     flag_handler = build_system_flags
 
     def url_for_version(self, version):
         return "https://github.com/ACCESS-NRI/GFDL-generic-tracers/tarball/{0}".format(version)
 
+    # TODO: We should try to remove this. The responsibility for including
+    #       internal library dependencies for linking purposes should
+    #       be in the source package's build system.
     @property
     def libs(self):
         libraries = find_libraries(
-            "libgtracers", root=self.prefix, shared=False, recursive=True
+            "libgtracers",
+            root=self.prefix,
+            shared=self.spec.variants.get("shared"),
+            recursive=True
         )
         # mocsy is an internal lib dependency of this static library.
         # Hence we need to propagate it.
         return libraries + self.spec["access-mocsy"].libs
+
+    def cmake_args(self):
+        return [self.define_from_variant("BUILD_SHARED_LIBS", "shared")]
```

Comment:
Just noting that in `access-fms` this is `SHARED_LIBS` and not `BUILD_SHARED_LIBS`.

I'm sure it's fine, but just wanted to note the inconsistency.

Reply:
CMake documentation uses `BUILD_SHARED_LIBS`: https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html